udp.go

```diff
@@ -9,6 +9,34 @@ import (
 	"golang.org/x/net/ipv6"
 )
 
+type SessionUDPFactory interface {
+	// InitConn set's up 'conn' to be used with a SessionUDP.
+	// Must be called before 'conn' is passed to SessionUDP.ReadRequest()
+	InitConn(conn *net.UDPConn) error
+
+	// CreateSessionUDP creates a SessionUDP object which can manage the state
+	// of a single UDP transaction at a time.
+	// Multiple SessionUDP objects may be created on the same underlying 'conn' by
+	// calling this multiple times with the same 'conn'.
+	CreateSessionUDP(msgSize int) SessionUDP
+}
+
+// SessionUDP holds manages a UDP Request/Response transaction.
+type SessionUDP interface {
+	// Clear re-initializes SessionUDP to the same state it was when new.
+	// This is required to enable pooling.
+	Clear() SessionUDP
+	// ReadRequest reads a single request from the UDPSession
+	ReadRequest(conn *net.UDPConn) ([]byte, error)
```

Comment:
It's really not very clean that this is the only method that takes a `*net.UDPConn` and it's supposed to stash it away for later use.